libs/utils/src/lib/services/broadcast-channel/broadcast-channel.service.ts

```diff
@@ -0,0 +1,95 @@
+import { isPlatformBrowser } from '@angular/common';
+import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
+import { EMPTY, fromEvent, Observable } from 'rxjs';
+
+@Injectable({
+	providedIn: 'root',
+})
+export class BroadcastChannelService {
+	private broadcastChannel: Record<string, BroadcastChannel> = {};
+
+	constructor(@Inject(PLATFORM_ID) private platformId: string) {}
+
+	/**
+	 * initChannel
+	 *
+	 * The initChannel method initializes a new BroadcastChannel instance.
+	 *
+	 * @param channelName{string} - The name of the Broadcast Channel.
+	 */
+	public initChannel(channelName: string): void {
```

Comment:
This is currently the only constructor argument in the spec but doing it this way isn't very future-proof

```suggestion
	public initChannel(...args: ConstructorParameters<typeof BroadcastChannel>): void {
```

<img width="473" alt="Screenshot 2024-10-17 at 09 28 27" src="https://github.com/user-attachments/assets/e9c2690c-d7a0-4e29-9cd4-841e36c16690">